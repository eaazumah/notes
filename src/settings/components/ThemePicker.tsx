import React from 'react';
import Modal from 'react-native-modal';
import styled from 'styled-components/native';
import useTheme from '../../@shared/hooks/useTheme';

interface Props {
  isVisible: boolean;
  setIsVisible?: (e: boolean) => void;
}

const ThemePicker: React.FC<Props> = props => {
  const {updateTheme} = useTheme();

  const changeToDark = () => {
    updateTheme('dark');
    props.setIsVisible?.(false);
  };

  const changeToLight = () => {
    updateTheme('light');
    props.setIsVisible?.(false);
  };

  const changeToSystem = () => {
    updateTheme('system');
    props.setIsVisible?.(false);
  };

  return (
    <Modal testID={'theme-picker'} isVisible={props.isVisible}>
      <Content>
        <Item onPress={changeToLight} testID={'light-item'}>
          <Text>Light</Text>
        </Item>
        <Item onPress={changeToDark} testID={'dark-item'}>
          <Text>Dark</Text>
        </Item>
        <Item onPress={changeToSystem} testID={'system-item'}>
          <Text>System Theme</Text>
        </Item>
      </Content>
    </Modal>
  );
};

const Text = styled.Text`
  color: ${props => props.theme.primaryText};
`;

const Content = styled.View`
  width: 100%;
  border-radius: 6px;
  align-items: center;
  justify-content: center;
  background-color: ${_ => _.theme.backgroundColor};
`;

const Item = styled.TouchableOpacity`
  width: 100%;
  height: 50px;
  align-items: center;
  justify-content: center;
  border-bottom-width: 1px;
  border-bottom-left-radius: 6px;
  border-bottom-right-radius: 6px;
  border-bottom-color: ${_ => _.theme.surface};
  background-color: ${_ => _.theme.backgroundColor};
`;

export default ThemePicker;
