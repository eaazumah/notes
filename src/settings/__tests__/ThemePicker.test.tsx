import {cleanup, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import TestProvider from '../../__testUtils__/Providers';
import ThemePicker from '../components/ThemePicker';

describe('Theme Picker', () => {
  afterEach(cleanup);
  test('Theme Picker renders correctly', () => {
    const setIsVisible = jest.fn();
    const screen = render(
      <TestProvider>
        <ThemePicker isVisible={false} setIsVisible={setIsVisible} />
      </TestProvider>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
    const component = screen.getByTestId('theme-picker');
    expect(component.props.isVisible).toEqual(false);
  });

  test('Theme Picker Select Dark Theme Functionality Test', () => {
    const setIsVisible = jest.fn();
    const screen = render(
      <TestProvider>
        <ThemePicker isVisible={true} setIsVisible={setIsVisible} />
      </TestProvider>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
    const modal = screen.getByTestId('theme-picker');
    expect(modal.props.isVisible).toEqual(true);
    const darkItemComponent = screen.getByTestId('dark-item');
    fireEvent.press(darkItemComponent);
    expect(setIsVisible).toBeCalledWith(false);
  });

  test('Theme Picker Select Light Theme Functionality Test', () => {
    const setIsVisible = jest.fn();
    const screen = render(
      <TestProvider>
        <ThemePicker isVisible={true} setIsVisible={setIsVisible} />
      </TestProvider>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
    const modal = screen.getByTestId('theme-picker');
    expect(modal.props.isVisible).toEqual(true);
    const lightItemComponent = screen.getByTestId('light-item');
    fireEvent.press(lightItemComponent);
    expect(setIsVisible).toBeCalledWith(false);
  });

  test('Theme Picker Select System Theme Functionality Test', () => {
    const setIsVisible = jest.fn();
    const screen = render(
      <TestProvider>
        <ThemePicker isVisible={true} setIsVisible={setIsVisible} />
      </TestProvider>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
    const modal = screen.getByTestId('theme-picker');
    expect(modal.props.isVisible).toEqual(true);
    const systemItemComponent = screen.getByTestId('system-item');
    fireEvent.press(systemItemComponent);
    expect(setIsVisible).toBeCalledWith(false);
  });
});
