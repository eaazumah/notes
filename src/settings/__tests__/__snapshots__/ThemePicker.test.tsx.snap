// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Picker Theme Picker Select Dark Theme Functionality Test 1`] = `
<Modal
  isVisible={true}
  testID="theme-picker"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderRadius": 6,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderBottomColor": "#e6e6e6",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderBottomWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="light-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#242A37",
            },
          ]
        }
      >
        Light
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderBottomColor": "#e6e6e6",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderBottomWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="dark-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#242A37",
            },
          ]
        }
      >
        Dark
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderBottomColor": "#e6e6e6",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderBottomWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="system-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#242A37",
            },
          ]
        }
      >
        System Theme
      </Text>
    </View>
  </View>
</Modal>
`;

exports[`Theme Picker Theme Picker Select Light Theme Functionality Test 1`] = `
<Modal
  isVisible={true}
  testID="theme-picker"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#151515",
          "borderRadius": 6,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#151515",
          "borderBottomColor": "#171c26",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderBottomWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="light-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        Light
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#151515",
          "borderBottomColor": "#171c26",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderBottomWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="dark-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        Dark
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#151515",
          "borderBottomColor": "#171c26",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderBottomWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="system-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        System Theme
      </Text>
    </View>
  </View>
</Modal>
`;

exports[`Theme Picker Theme Picker Select System Theme Functionality Test 1`] = `
<Modal
  isVisible={true}
  testID="theme-picker"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderRadius": 6,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderBottomColor": "#e6e6e6",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderBottomWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="light-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#242A37",
            },
          ]
        }
      >
        Light
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderBottomColor": "#e6e6e6",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderBottomWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="dark-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#242A37",
            },
          ]
        }
      >
        Dark
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderBottomColor": "#e6e6e6",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderBottomWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="system-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#242A37",
            },
          ]
        }
      >
        System Theme
      </Text>
    </View>
  </View>
</Modal>
`;

exports[`Theme Picker Theme Picker renders correctly 1`] = `
<Modal
  isVisible={false}
  testID="theme-picker"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderRadius": 6,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderBottomColor": "#e6e6e6",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderBottomWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="light-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#242A37",
            },
          ]
        }
      >
        Light
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderBottomColor": "#e6e6e6",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderBottomWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="dark-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#242A37",
            },
          ]
        }
      >
        Dark
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderBottomColor": "#e6e6e6",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderBottomWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="system-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#242A37",
            },
          ]
        }
      >
        System Theme
      </Text>
    </View>
  </View>
</Modal>
`;
